backend/src/org/commcare/xml/EntryParser.java

```diff
@@ -18,22 +20,25 @@
 /**
  * @author ctsims
  */
-public class EntryParser extends CommCareElementParser<Entry> {
-    boolean isEntry = true;
+public class EntryParser extends CommCareElementParser<EntryBase> {
+    private boolean isEntry = true;
 
-    public EntryParser(KXmlParser parser) {
-        this(parser, true);
-    }
-
-    public EntryParser(KXmlParser parser, boolean isEntry) {
+    private EntryParser(KXmlParser parser, boolean isEntry) {
         super(parser);
+
         this.isEntry = isEntry;
     }
 
-    /* (non-Javadoc)
-     * @see org.javarosa.xml.ElementParser#parse()
-     */
-    public Entry parse() throws InvalidStructureException, IOException, XmlPullParserException {
+    public static EntryParser buildViewParser(KXmlParser parser) {
+        return new EntryParser(parser, false);
+    }
+
+    public static EntryParser buildEntryParser(KXmlParser parser) {
+        return new EntryParser(parser, true);
+    }
+
+    @Override
+    public EntryBase parse() throws InvalidStructureException, IOException, XmlPullParserException {
         String block = isEntry ? "entry" : "view";
```

Comment:
Views have existed for a while, I just made them explicit classes instead of a distinction derived from the internal state of an `Entry` object (I think when the namespace is `null`)